dimscord/gateway.nim

```diff
@@ -371,23 +371,23 @@ proc handleDispatch(s: Shard, event: string, data: JsonNode) {.async.} =
                 emojis.add(newEmoji(emji))
                 g.emojis.add(emji["id"].str, newEmoji(emji))
 
-            cl.events.guild_emojis_update(s, g, emojis)
+            await cl.events.guild_emojis_update(s, g, emojis)
         of "USER_UPDATE":
             let user = newUser(data)
             var oldUser: Option[User] = none(User)
             cl.user = user
 
-            cl.events.user_update(s, user, oldUser)
+            await cl.events.user_update(s, user, oldUser)
         of "PRESENCE_UPDATE":
             var oldPresence: Option[Presence] = none(Presence)
             var presence = newPresence(data)
-            if cl.cache.preferences.cache_guilds:
+            if cl.cache.preferences.cache_guilds and presence.guild_id in cl.cache.guilds:
                 let guild = cl.cache.guilds[presence.guild_id]
 
                 if guild.presences.hasKey(presence.user.id):
                     oldPresence = some(guild.presences[presence.user.id])
 
-                var member = guild.members[presence.user.id]
+                var member = guild.members.getOrDefault(presence.user.id)

```

Comment:
and here if it doesn't exist we get the default value for the member, not sure if that's correct but at least it doesn't crash :D